src/webgpu/api/validation/encoding/queries/begin_end.spec.ts

```diff
@@ -14,45 +14,91 @@ TODO:
         - }
 `;
 
+import { pbool } from '../../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
 import { ValidationTest } from '../../validation_test.js';
 
 export const g = makeTestGroup(ValidationTest);
 
-g.test('occlusion_query,begin_end')
+g.test('occlusion_query,begin_end_balance')
   .desc(
     `
-Tests that begin/end occlusion query on render pass:
-- begin 0
-- begin 0, end (control case)
-- begin 0, begin 1, end
-- begin 0, end, end
-- end
+Tests that begin/end occlusion queries mismatch on render pass:
+- begin 0, end 1 (begin no queries, then end one query)
+- begin 1, end 0 (begin one query, then end no queries)
+- begin 1, end 1 (begin one query, then end one query)(control case)
+- begin 1, end 2 (begin one query, then end two queries)
+- begin 2, end 1 (begin two queries, then end one query)
   `
   )
+  .subcases(
+    () =>
+      [
+        { begin: 0, end: 1 },
+        { begin: 1, end: 0 },
+        { begin: 1, end: 1 },
+        { begin: 1, end: 2 },
+        { begin: 2, end: 1 },
+      ] as const
+  )
+  .unimplemented();
+
+g.test('occlusion_query,begin_end_invalid_nesting')
+  .desc(
+    `
+Tests the invalid nesting of begin/end occlusion queries:
+- begin index 0, begin index 0, end, end
+- begin index 0, begin index 1, end, end
+  `
+  )
+  .subcases(() => [{ calls: [0, 0, 'end', 'end'] }, { calls: [0, 1, 'end', 'end'] }] as const)
```

Comment:
If neither of these is valid (I'm not sure), there should be some control cases here like `[0, 'end']`, `[0, 'end', 1, 'end']` which would be valid.